### PR TITLE
Don't re-export `modifyMaxSize`.

### LINKED ISSUE #22: I think `modifyMaxSize` has no effect
This function is re-exported from Test.Hspec.QuickCheck, but it looks to me like it has (and can have) no effect. `maxSize` of the quickcheck args isn't used, and I believe Hedgehog currently has no way to set the max size for a test. I guess it would be worth documenting this, and not re-exporting the function.

### PR DESCRIPTION
**Pull request by ChickenProp**
Closes #22.

### CODE REVIEW

File: src/Test/Hspec/Hedgehog.hs [hunk 1]
```diff
@@ -84,10 +84,12 @@ module Test.Hspec.Hedgehog
     ( -- * The Main Function
       hedgehog
       -- * Hspec re-exports
+      --
+      -- | 'Test.Hspec.QuickCheck.modifyMaxSize' isn't re-exported, since
+      -- hedgehog has nothing that corresponds to it.
```
Comment (sol) [inline src/Test/Hspec/Hedgehog.hs:89]: I am 50/50 on this one.  Is this useful information for someone who is going to use the API? Or is this just the justification for us to remove the re-export? If it's the later, then that piece of documentation can be removed.

@ChickenProp I defer to your judgement.

**Review by sol — CHANGES_REQUESTED** [review]
This looks good.

1. Regarding the documentation, I defer to your judgement.
2. Rebase on `main`.
3. Add change log entry.
4. If you think with this PR current `main` will be ready for a release, then please also bump the version.  Releases happen automatically on version bumps.

File: src/Test/Hspec/Hedgehog.hs (hunk 1 above)
Comment (ChickenProp) [reply src/Test/Hspec/Hedgehog.hs:89]: I think useful, yeah. I've used this re-export in the past, not realizing it was useless. If it had been removed without saying why in the docs, I would have either spent a while trying to figure out why it was missing, or just taken the function from its original location.

**Comment by ChickenProp** [comment]
Done.

**Review by sol — APPROVED** [review]

**Pull request merged**